doc/02_user/integration.rst

```diff
@@ -28,6 +28,63 @@ The vendor ID and implementation ID (``mvendorid`` and ``mimpid``) both read as
 Implementers may wish to use other values here.
 Please see the RISC-V Privileged Architecture specification for more details on what these IDs represent and how they should be chosen.
 
+.. _integration-prims:
+
+Primitives
+----------
+
+Ibex uses a number of primitive modules (that are held outside the :file:`rtl/` which contains the Ibex RTL).
+Full implementations of these primitives are provided in the Ibex repository but implementors may wish to provide their own implementations.
+Some of the primitives are only used for specific Ibex configurations so can be ignored/removed if you're not using one of those configurations.
+
+The mandatory primitives (used by all configurations) are:
+ * ``prim_buf`` - A buffer, used to ensure security critical logic isn't optimized out in synthesis (by applying suitable constraints to prim_buf).
+   In configurations where ``SecureIbex == 0`` it must exist but can be implemented as a straight passthrough.
+ * ``prim_clock_gating`` - A clock gate.
+
+The configuration dependent primitives are:
+ * ``prim_clock_mux2`` - A clock mux, used by the lockstep duplicate core.
+   Required where ``SecureIbex == 1``.
+ * ``prim_flop`` - A multi-bit flop, , used to ensure security critical logic isn't optimized out in synthesis (by applying suitable constraints to prim_flop).
```

Comment:
I find "multi-bit flop" a bit confusing here because most of these prims are potentially multiple bits wide. I would just write "A flip flop".